sql/core/src/main/scala/org/apache/spark/sql/expressions/UserDefinedFunction.scala

```diff
@@ -103,4 +110,19 @@ case class UserDefinedFunction protected[sql] (
       udf
     }
   }
+
+  /**
+   * Updates UserDefinedFunction to non-deterministic.
+   *
+   * @since 2.3.0
+   */
+  def nonDeterministic(): UserDefinedFunction = {
```

Comment:
Originally, this is from @rxin. I am fine for both ways. If needed, we can have both?